README.md

```diff
@@ -7,3 +7,194 @@ Status](https://github.com/maltezfaria/DataFlowTasks.jl/workflows/CI/badge.svg)]
 [![codecov](https://codecov.io/gh/maltezfaria/DataFlowTasks.jl/branch/main/graph/badge.svg?token=UOWU691WWG)](https://codecov.io/gh/maltezfaria/DataFlowTasks.jl)
 ![Lifecycle](https://img.shields.io/badge/lifecycle-experimental-blue.svg)
 
+`DataFlowTasks.jl` is a Julia package dedicated to parallel programming on multi-core shared memory CPUs. From user annotations (READ, WRITE, READWRITE) on program data, `DataFlowTasks.jl` automatically infers dependencies between parallel tasks.
+
+The usual linear algebra data types (Julia arrays) are particularly easy to use with `DataFlowTasks.jl`.
+
+## Installation
+
+```julia
+using Pkg
+Pkd.add("https://github.com/maltezfaria/DataFlowTasks.jl.git")
+```
+
+## Basic Usage
+
+The use of a `DataFlowTask`s is intended to be as similar to a Julia native `Task`s as possible. The API implements these three macros :
+* `@dspawn`
+* `@dtask`
+* `@dasync`
+
+which behaves like there `Base` counterparts, except they need additional annotations to specify access modes. This is done with the three macros :
+* `@R`
+* `@W`
+* `@RW`
+
+where, in a function argument or at the beginning of a task block, `@R(A)` implies that A will be in read mode in the function/block.
+
+Let's look at how it can parallelize with safety insurance.
+
+```julia
+write!(X, alpha) = (X .= alpha)
+readwrite!(X) = (X .+= 2)
+
+n = 1000
+A = ones(n)
+
+@dspawn write!(@R(A), 1)          # 1
+@dspawn write!(@R(A[1:500]), 2)   # 2
+@dspawn write!(@R(A[501:n]), 3)   # 3
+@dspawn begin                     # 4
+  @RW A
+  readwrite!(A)
+end
+
+DataFlowTasks.sync()
+```
+
+This will generate the DAG (Directed Acyclic Graph) below that represents the dependencies between tasks. This means that the task 2 and 3 can be run in parallel. We see how it's the memory that matters here.
+
+![Basic](graph.png)
```

Comment:
Are you sure that this is the `DAG` generated? I seems like tasks 2 and 3 operate on new data, and not on a view of `A`, so I would suspect that they are independent of everything else.

Reply:
It isn't indeed, using views does work. It's related to the issue mentioned yesterday. I would expect this behaviour to work though, couldn't it ? Is it also related to the Issue #25 that François highlighted, about the let blocks ?